src/planner/expression/bound_cast_expression.cpp

```diff
@@ -149,10 +149,14 @@ bool BoundCastExpression::CastIsInvertible(const LogicalType &source_type, const
 	case LogicalTypeId::TIMESTAMP_MS:
 	case LogicalTypeId::TIMESTAMP_NS:
 		switch (target_type.id()) {
+		case LogicalTypeId::TIMESTAMP:
```

Comment:
I don't think this is quite right - TS => TS_SEC _is_ invertible.

Reply:
Sorry I meant that _all_ of the downcasts are reversible!

Reply:
I think I have it correct now, assuming the `TypeId`s in types.hpp are respect significance in an increasing order. `TimeStamp` and `Timestamp_ns` might need another look though